qcodes/instrument_drivers/devices.py

```diff
@@ -1,6 +1,190 @@
+        self._wrapper_param = output
+
+        # Set the role, either as divider or amplifier
+        # Raise an error if nothing is specified
+        is_divider = division is not None
+        is_amplifier = gain is not None
+
+        if not (is_divider ^ is_amplifier):
+            raise ValueError('Provide only division OR gain')
+
+        if is_divider:
+            self.role = Role.DIVISION
+            self._multiplier = division
+        elif is_amplifier:
+            self.role = Role.GAIN
+            self._multiplier = gain
+
+        # Set the name
+        if name:
+            self.name = name
+        else:
+            if self.role == Role.DIVISION:
+                self.name = "{}_attenuated".format(self._wrapper_param.name)
+            elif self.role == Role.GAIN:
+                self.name = "{}_amplified".format(self._wrapper_param.name)
+
+        # Set label
+        if label:
+            self.label = label
+        elif name:
+            self.label = name
+        else:
+            if self.role == Role.DIVISION:
+                self.label = "{}_attenuated".format(self._wrapper_param.label)
+            elif self.role == Role.GAIN:
+                self.label = "{}_amplified".format(self._wrapper_param.label)
+
+        # Set the unit
+        if unit:
+            self.unit = unit
+        else:
+            self.unit = self._wrapper_param.unit
+
+        super().__init__(
+            name=self.name,
+            instrument=getattr(self._wrapper_param, "_instrument", None),
+            label=self.label,
+            unit=self.unit,
+            metadata=self._wrapper_param.metadata)
+
+        # extend metadata
+        self._meta_attrs.extend(["division"])
+        self._meta_attrs.extend(["gain"])
+
+    # Internal handling of the multiplier
+    # can be either a Parameter or a scalar
+    @property
+    def _multiplier(self):
+        return self._multiplier_parameter
+
+    @_multiplier.setter
+    def _multiplier(self, multiplier: Union[int, float, Parameter]):
+        if isinstance(multiplier, Parameter):
+            self._multiplier_parameter = multiplier
+        else:
+            self._multiplier_parameter = ManualParameter(
+                'multiplier', initial_value=multiplier)
+
+    # Division of the scaler
+    @property
+    def division(self):
+        if self.role == Role.DIVISION:
+            return self._multiplier()
+        elif self.role == Role.GAIN:
+            return 1 / self._multiplier()
+
+    @division.setter
+    def division(self, division: Union[int, float, Parameter]):
+        self.role = Role.DIVISION
+        self._multiplier = division
+
+    # Gain of the scaler
+    @property
+    def gain(self):
+        if self.role == Role.GAIN:
+            return self._multiplier()
+        elif self.role == Role.DIVISION:
+            return 1 / self._multiplier()
+
+    @gain.setter
+    def gain(self, gain: Union[int, float, Parameter]):
+        self.role = Role.GAIN
```

Comment:
Do you see any use in changing the role *after* the ScalerParameter object is created? I personally don’t because it may lead to errors in users code (also when the role is changed, labels and names are not updated) - thus, I’d suggest that if the role is “gain”, it should not be possible to set the “division” and vice versa.

Reply:
On the contrary, I found very useful sometimes when changing the gain, to change also the way how's expressed (division or gain). Also some instruments have a variable gain that can be either less or more then one. I would rather remove the _attenuated/_amplified naming and use only "_scaled"